File: Examples/Ruby/Sections/update_section_page_setup.rb
require 'aspose_words_cloud'

class WorkingWithPageSetup

  include AsposeWordsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @words_api = WordsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def update_section_page_setup
    file_name = "SampleWordDocument.docx"
    upload_file(file_name)

    page_setup = PageSetup.new
    page_setup.rtl_gutter = true
    page_setup.left_margin = 10.0
    page_setup.orientation = "Landscape"
    page_setup.paper_size = "A5"

    section_index = 0
    response = @words_api.update_section_page_setup(file_name, section_index, page_setup)
  end

end

workingWithPageSetup = WorkingWithPageSetup.new()
puts workingWithPageSetup.update_section_page_setup

File: Examples/PHP/Sections/UpdateSectionPageSetup.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Words\WordsApi;
use Aspose\Words\AsposeApp;

class Sections {

	public $wordsApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->wordsApi = new WordsApi();
	}


	public function updateSectionPageSetup() {
		// Upload file to Aspose Cloud Storage
		$fileName = "SampleWordDocument.docx";
		Utils::uploadFile($fileName);

        $body = '{
                    "PageSetup": {
                        "LeftMargin":10.0,
                        "Orientation": "Landscape",
                        "PaperSize": "A5"
                    }
                 }';
        $result = $this->words->UpdateSectionPageSetup($fileName, $sectionIndex=0, $storage = null, $folder = null, $filename = null, $body);
        print_r($result);
    }
}

$sections = new Sections();
$sections->updateSectionPageSetup();

?>
//ExEnd:
